File: Inverted-Index/src/tests/DirectoryReaderTest.java
package tests;

import java.io.File;
import org.junit.Test;

import exceptions.NotDirectoryException;

import org.junit.Assert;
import org.junit.Before;

import file_handler.DirectoryReader;

public class DirectoryReaderTest {

    private final int FILES_COUNT = 3;
    private final String[] TEST_FILE_NAMES = new String[] {"57110", "59483", "59519"};
    private final String TEST_DOCS_PATH = "C:\\Repos\\Summer1400-SE-Team24\\Inverted-Index\\src\\tests\\docs";
    private File[] files;

    @Before
    public void readingFiles() throws NotDirectoryException {
        var directoryReader = new DirectoryReader();
        files = directoryReader.getFiles(TEST_DOCS_PATH);
    }

    @Test
    public void readedFilesCountTest() {

        Assert.assertEquals(FILES_COUNT, files.length);
    }

    @Test
    public void readedFilesNamesTest() {
        for (int i = 0; i < TEST_FILE_NAMES.length; i++) {
            Assert.assertEquals(TEST_FILE_NAMES[i], files[i].getName());
        }
    }
}

File: Phase5-TDD/InvertedIndex.Test/DirectoryReaderTest.cs
using System.IO;
using System.Linq;
using Xunit;

using InvertedIndex.FileHandler;

namespace InvertedIndex.Test
{
    public class DirectoryReaderTest
    {
        private const int FILES_COUNT = 3;
        private readonly string[] TEST_FILE_NAMES = new string[] {"57110","59483","59519"};
        private readonly string TEST_DOCS_PATH = Path.GetFullPath("../../../docs/");
        private FileInfo[] _files;

        public DirectoryReaderTest()
        {
            var directoryReader = new DirectoryReader();
            _files = directoryReader.GetFiles(TEST_DOCS_PATH).OrderBy(f => f.Name).ToArray();
        }

        [Fact]
        public void ReadFilesCountTest()
        {
            Assert.Equal(FILES_COUNT, _files.Length);
        }

        [Fact]
        public void ReadFilesNamesTest()
        {
            for (int i = 0; i < TEST_FILE_NAMES.Length; i++)
            {
                Assert.Equal(TEST_FILE_NAMES[i], _files[i].Name);
            }
        }
    }
}